### ISSUE TITLE
Support Scala 2.12

### ISSUE THREAD

**Issue by zouzias** [zouzias/spark-lucenerdd#176]
**Is your feature request related to a problem? Please describe.**

Spark supports 2.12 without experimental tag since 2.4.1. It would be nice to support it as well here.

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Blockers are:
* spark-testing-base does not support 2.12 yet.

**Comment by icassina** [comment]
I'm experimenting with this. I'm just missing `EmptyMap` and `EmptySet` in KryoRegistrators.
Testing with our code base when I manage to resolve other missing dependencies for scala 2.12

**Comment by saumyasuhagiya** [comment]
Any plan on this?

**Comment by zouzias** [comment]
Hi @saumyasuhagiya , this is done in develop after upgrading to Spark 3.x

I just released a snapshot version `0.4.0-SNAPSHOT`

```
[info] Main Scala API documentation successful.
[warn] so tooling can use it for eviction errors etc - https://www.scala-sbt.org/1.x/docs/Publishing.html
[info] 	published spark-lucenerdd_2.12 to https://oss.sonatype.org/content/repositories/snapshots/org/zouzias/spark-lucenerdd_2.12/0.4.0-SNAPSHOT/spark-lucenerdd_2.12-0.4.0-SNAPSHOT.jar.asc
[info] 	published spark-lucenerdd_2.12 to https://oss.sonatype.org/content/repositories/snapshots/org/zouzias/spark-lucenerdd_2.12/0.4.0-SNAPSHOT/spark-lucenerdd_2.12-0.4.0-SNAPSHOT-sources.jar
[info] 	published spark-lucenerdd_2.12 to https://oss.sonatype.org/content/repositories/snapshots/org/zouzias/spark-lucenerdd_2.12/0.4.0-SNAPSHOT/spark-lucenerdd_2.12-0.4.0-SNAPSHOT-sources.jar.asc
[info] 	published spark-lucenerdd_2.12 to https://oss.sonatype.org/content/repositories/snapshots/org/zouzias/spark-lucenerdd_2.12/0.4.0-SNAPSHOT/spark-lucenerdd_2.12-0.4.0-SNAPSHOT.pom.asc
[info] 	published spark-lucenerdd_2.12 to https://oss.sonatype.org/content/repositories/snapshots/org/zouzias/spark-lucenerdd_2.12/0.4.0-SNAPSHOT/spark-lucenerdd_2.12-0.4.0-SNAPSHOT-javadoc.jar.asc
[info] 	published spark-lucenerdd_2.12 to https://oss.sonatype.org/content/repositories/snapshots/org/zouzias/spark-lucenerdd_2.12/0.4.0-SNAPSHOT/spark-lucenerdd_2.12-0.4.0-SNAPSHOT-javadoc.jar
[info] 	published spark-lucenerdd_2.12 to https://oss.sonatype.org/content/repositories/snapshots/org/zouzias/spark-lucenerdd_2.12/0.4.0-SNAPSHOT/spark-lucenerdd_2.12-0.4.0-SNAPSHOT.jar
[info] 	published spark-lucenerdd_2.12 to https://oss.sonatype.org/content/repositories/snapshots/org/zouzias/spark-lucenerdd_2.12/0.4.0-SNAPSHOT/spark-lucenerdd_2.12-0.4.0-SNAPSHOT.pom
[success] Total time: 14 s, completed Aug 25, 2021 9:23:33 AM

```

**Comment by zouzias** [comment]
> I'm experimenting with this. I'm just missing `EmptyMap` and `EmptySet` in KryoRegistrators.
> Testing with our code base when I manage to resolve other missing dependencies for scala 2.12

Thanks for your input. @icassina, Is there a better way to add all the classes in the KryoRegistrators? I find it annoying having to register those classes manually each time.
